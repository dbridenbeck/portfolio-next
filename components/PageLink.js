import { motion, AnimatePresence } from "framer-motion";

const variants = {
  moveUp: { opacity: 0, y: "100px", x: "0px" },
  center: { opacity: 1, y: "0", x: "0px", rotate: 360 },
  moveRight: { opacity: 0, y: "0px", x: "100px" },
};

const PageLink = ({ onHelloPage }) => (
  <>
    <AnimatePresence initial={false}>
      {onHelloPage && (
        <>
          <span className="pageLink">
            <motion.div
              className="linkText"
              initial="moveUp"
              animate="center"
              exit="moveRight"
              transition={{ duration: 1.5 }}
              variants={variants}
            >
              about
            </motion.div>
          </span>
        </>
      )}
    </AnimatePresence>
    <AnimatePresence initial={false}>
      {!onHelloPage && (
        <span className="pageLink">
          <motion.div
            className="linkText"
            initial="moveUp"
            animate="center"
            exit="moveRight"
            transition={{ duration: 1.5 }}
            variants={variants}
          >
            hello
          </motion.div>
        </span>
      )}
    </AnimatePresence>

    <style jsx>{`
      .pageLink {
        position: absolute;
        display: block;
        right: 60px;
        bottom: 90px;
        font-size: 0.9em;
        font-style: italic;
        color: ${onHelloPage ? "#8affff" : "#c5ff8a"};
        cursor: pointer;
      }

      .linkText {
        position: absolute;
        display: inline;
        z-index: 99;
      }
    `}</style>
  </>
);

export default PageLink;