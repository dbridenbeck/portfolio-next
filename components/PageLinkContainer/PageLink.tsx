import { AnimatePresence } from "framer-motion";
import PageLinkModel from "../../models/pageLink";
import PageLinkAnimation from "../../animations/PageLinkAnimation";

const PageLink: React.FC<PageLinkModel> = ({ onHelloPage }) => (
  <>
    {/* 
      onHelloPage & !onHelloPage used to add/remove from DOM
      which triggers AnimatePresence to fire animation
    */}
    <AnimatePresence initial={false}>
      {onHelloPage && (
        <span className="pageLink blue">
          <PageLinkAnimation>about</PageLinkAnimation>
        </span>
      )}
    </AnimatePresence>
    <AnimatePresence initial={false}>
      {!onHelloPage && (
        <span className="pageLink red">
          <PageLinkAnimation>hello</PageLinkAnimation>
        </span>
      )}
    </AnimatePresence>

    <style jsx>{`
      .pageLink {
        position: absolute;
        display: block;
        right: 60px;
        font-size: 0.9em;
        font-style: italic;
        cursor: pointer;
      }

      .linkText {
        position: absolute;
        display: inline;
        z-index: 99;
      }
    `}</style>
  </>
);

export default PageLink;