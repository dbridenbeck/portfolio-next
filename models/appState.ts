export interface ProjectModel {
  title: string;
  type: string;
  tech: string[];
  projectDetail: string;
  url: string;
  color: string;
  webm: string;
  mp4: string;
  open: boolean;
  thumb: string;
  thumbAlt: string;
}

export default interface AppStateModel {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  projects: ProjectModel[];
  pageClickedOnce: boolean;
}
