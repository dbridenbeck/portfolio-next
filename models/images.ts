export default interface ImageModel {
  currentPage: string;
  pageClickedOnce: boolean;
}