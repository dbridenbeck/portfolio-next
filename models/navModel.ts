export interface NavProps {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  changePage: (text: string) => void;
}
