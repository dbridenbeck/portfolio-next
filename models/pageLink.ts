export default interface PageLink {
  onHelloPage: boolean;
}
