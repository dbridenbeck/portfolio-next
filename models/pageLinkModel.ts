export interface PageLinkProps {
  page: { pageName: string; color: string };
  currentPage: string;
  handleChangePage: (text: string) => void;
}
