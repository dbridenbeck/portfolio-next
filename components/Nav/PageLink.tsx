import styled from 'styled-components';
import { PageLinkProps } from "../../models/pageLinkModel";

const PageLinkSpan = styled.span`
  display: block;
  font-size: 1.125em;
  cursor: pointer;
  color: ${({ linkSelected, color }) => linkSelected ? color : "black"};
  :hover {
  color: ${({ color }) => color};
  }
`;

const PageLink: React.FC<PageLinkProps> = ({page, handleChangePage, currentPage}) => {
  const linkSelected = currentPage === page.pageName;
  const updatePage = () => handleChangePage(page.pageName);
  return (
    <>
      <PageLinkSpan 
        color={page.color}
        linkSelected={linkSelected}
        onClick={updatePage}
      >
        {page.pageName}
      </PageLinkSpan>
    </>
  )
};

export default PageLink;