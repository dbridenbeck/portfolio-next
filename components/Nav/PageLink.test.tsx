import React from 'react';
import { render } from '@testing-library/react';
import PageLink from './PageLink';
import { PageLinkProps } from "../../models/pageLinkModel";

describe("PageLink", () => {
  const requiredProps: PageLinkProps = {
    page: {
      pageName: "skills",
       color: "#6f59e9"
      },
    currentPage: "about",
    handleChangePage: jest.fn(),
  };
  
  it("should render without errors", () => {
    const { container } = render(<PageLink {...requiredProps} />);
    expect(container).toMatchSnapshot();
  }); 
});