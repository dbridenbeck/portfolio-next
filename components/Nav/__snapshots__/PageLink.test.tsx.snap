// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLink should render without errors 1`] = `
<div>
  <span
    class="PageLink__PageLinkSpan-x2cztj-0 kGmCBl"
    color="#6f59e9"
  >
    skills
  </span>
</div>
`;
